src/Interpreters/ActionsDAG.cpp

```diff
@@ -3838,4 +3914,12 @@ ActionsDAG ActionsDAG::deserialize(ReadBuffer & in, DeserializedSetsRegistry & r
     return dag;
 }
 
+bool ActionsDAG::containsNode(const Node * node)
```

Comment:
I prefer to avoid such an O(n) lookups.
It's only used once now, so probably we can do it there